discord/user.py

```diff
@@ -51,42 +68,42 @@ class BaseUser(_UserTag):
         bot: bool
         system: bool
 
-    def __init__(self, *, state, data):
-        self._state = state
+    def __init__(self, *, state: ConnectionState, data: UserPayload) -> None:
+        self._state: ConnectionState = state
         self._update(data)
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return (
             f"<BaseUser id={self.id} name={self.name!r} discriminator={self.discriminator!r}"
             f" bot={self.bot} system={self.system}>"
         )
 
-    def __str__(self):
+    def __str__(self) -> str:
         return f'{self.name}#{self.discriminator}'
 
-    def __eq__(self, other):
+    def __eq__(self, other: Any) -> bool:
         return isinstance(other, _UserTag) and other.id == self.id
 
-    def __ne__(self, other):
+    def __ne__(self, other: Any) -> bool:
         return not self.__eq__(other)
 
-    def __hash__(self):
+    def __hash__(self) -> int:
         return self.id >> 22
 
-    def _update(self, data):
-        self.name = data['username']
-        self.id = int(data['id'])
-        self.discriminator = data['discriminator']
-        self._avatar = data['avatar']
-        self._banner = data.get('banner', None)
-        self._accent_colour = data.get('accent_color', None)
-        self._public_flags = data.get('public_flags', 0)
-        self.bot = data.get('bot', False)
-        self.system = data.get('system', False)
+    def _update(self, data: UserPayload) -> None:
+        self.name: str = data['username']
```

Comment:
Either the `if TYPE_CHECKING` block above should be removed, or the types covered in that block should be removed here.